Build fails with rustc 1.80.1

**Describe the bug**
```
error[E0282]: type annotations needed
   --> src/ui.rs:388:26
    |
388 |                         .collect::<Box<_>>()
    |                          ^^^^^^^ cannot infer type of the type parameter `B` declared on the method `collect`
389 |                         .join(" ")
    |                          ---- type must be known at this point
    |
help: consider specifying the generic argument
    |
388 |                         .collect::<Vec<_>>()
    |                                 ~~~~~~~~~~

For more information about this error, try `rustc --explain E0282`.
error: could not compile `nvim-gtk` (bin "nvim-gtk") due to 1 previous error
```

Seems to be related to https://github.com/rust-lang/rust/issues/127343

Possible fix:
```diff
diff -ru a/src/ui.rs b/src/ui.rs
--- a/src/ui.rs
+++ b/src/ui.rs
@@ -385,7 +386,7 @@
                     files_list
                         .iter()
                         .map(|f| misc::escape_filename(f))
-                        .collect::<Box<_>>()
+                        .collect::<Box<[_]>>()
                         .join(" ")
                 ));
             } else {
```

**Technical information (please complete the following information):**
 - OS: NixOS unstable
 - Neovim version: 0.10.1
 - Neovim-Gtk build version: 1.0.4

--- a/pkgs/by-name/ne/neovim-gtk/collect-box.patch
+++ b/pkgs/by-name/ne/neovim-gtk/collect-box.patch
@@ -1,0 +1,12 @@
+diff -ru a/src/ui.rs b/src/ui.rs
+--- a/src/ui.rs
++++ b/src/ui.rs
+@@ -385,7 +386,7 @@
+                     files_list
+                         .iter()
+                         .map(|f| misc::escape_filename(f))
+-                        .collect::<Box<_>>()
++                        .collect::<Box<[_]>>()
+                         .join(" ")
+                 ));
+             } else {

--- a/pkgs/by-name/ne/neovim-gtk/package.nix
+++ b/pkgs/by-name/ne/neovim-gtk/package.nix
@@ -26,6 +26,8 @@ rustPlatform.buildRustPackage rec {
 
   buildInputs = [ gdk-pixbuf gtk4 pango vte-gtk4 ];
 
+  patches = [ ./collect-box.patch ];
+
   postInstall = ''
     make PREFIX=$out install-resources
   '';

